### PR TITLE
programs.khal: remove extra new line on config

### PR DESCRIPTION
**Pull request by octvs**
- Remove mistaken new line in khal config, refer to [1].
- Make necessary changes on the expected config files

[1]: https://github.com/nix-community/home-manager/pull/5192#pullrequestreview-1975091763

### Description

Refer to the commit msg.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- ~~If this PR adds a new module~~

  - [ ] ~~Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).~~

#### Maintainer CC

@teto

### CODE REVIEW

**Comment by teto** [comment]
thanks a lot for cleaning up my mess <3 
I must confess I am a bit surprised to see a contribution just to cleanup newlines (even though I am guilty of this myself but for systemd units, not generated files).
I can merge this as it is or if you are interested, you could add "addresses" support https://github.com/nix-community/home-manager/pull/5192 which would make home-manager support all the account-specific options of khal.

**Comment by octvs** [comment]
> I must confess I am a bit surprised to see a contribution just to cleanup newlines 
I could say that I was already "warm", working on a small patch to khard
module. Which got me excited to learn how to contribute to a project I
appreciate. 

> you could add "addresses" support
Thanks for the pointer, I thought it was a new module here on home-manager :D. 

Checking the khal documentation, I think it is not related to "birthday"
calendars [^1], since they wouldn't have event invites, I guess... Or the
approach here should be to not impose such deductions but let anything possible
on upstream to be possible through these modules.

Anyhow I will add the feature in a bit, I already sketched something.

In case it isn't obvious (/s), I'm quite new to this game, any feedback &
patience shown is appreciated.

[^1]: thus should be implemented for `accounts.calendar.accounts.<>.khal.` not
the `accounts.contact.accounts.<>.khal`

**Comment by octvs** [comment]
I've just added another commit for the aforementioned feature, once approved I
can squeeze them to a single commit if it is required.

File: modules/programs/khal-accounts.nix [hunk 1]
```diff
@@ -31,5 +31,15 @@ with lib;
...
       '';
     };
+
+    addresses = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = ''
+        Email addresses to be associated with this account. Used to check the
+        participation status ("PARTSTAT"), refer to khal documentation.
+
```
Comment (rycee) [inline modules/programs/khal-accounts.nix:41]:
```suggestion
```

**Comment by rycee** [comment]
LGTM but I'll let @teto have the last word 🙂

**Comment by octvs** [comment]
Just removed the emtpy line you suggested, thanks for the review.

When @teto gives his opinion I would squash all to a single commit.

**Comment by octvs** [comment]
Thank you for the kind compliment :).

> formatting that is off as the CI tells you. 
Oh I've missed that completely, that is to due to my globally set formatter
getting the upper hand via format-on-save, now should be fixed. 

>  home-manager project forces the squash so we dont even have a choice. 
I can't say I'm 100% clear on this, I would probably grasp the workflow more in
time. For now sticking with habits, I squash to clear the commit msgs,
meanwhile also rebased to the current HEAD. Sorry for triggering another CI run
here, especially if it is bad practice.

Thanks to both of you, for the review & your time.

**Comment by teto** [comment]
thank you for agreeing to my unsreasonable requests and welcome to home-manager ;)

**Comment by malikwirin** [comment]
> I can merge this as it is or if you are interested, you could add "addresses" support #5192 which would make home-manager support all the account-specific options of khal.

I am using this option like this:
```nix
{ config, ... }:

let
  emailCfg = config.accounts.email;
  base = emailAcc: {
    khal = {
      enable = true;
      addresses = [ emailAcc.address ];
    };
  };
in {
  accounts.calendar = {
    basePath = "${config.xdg.dataHome}/calendar";
    accounts = {
      default = base emailCfg.accounts.default // {
        primary = true;
      };
    };
  };
}
```
Now most khal commands through this warning:
```console
warning: unknown key or subsection "addresses" in section "[calendars][[default]]"
```
Is the khal version supporting this version not yet on 24.11?

**Comment by octvs** [comment]
> Is the khal version supporting this version not yet on 24.11?

I can't seem to reproduce your issue

```sh
$ cat ~/.config/khal/config | grep addresses
addresses= <redacted>
$ nix shell github:NixOS/nixpkgs/24.11#khal
$ khal --version
khal, version 0.11.3
$ khal at
$ khal printcalendars
... # executes succesfully
```

At first glance your problem stands unrelated to this issue. In any case, it is
a better idea to post this question with the resulting `~/.config/khal/config`
and the `khal --version`.

**Comment by malikwirin** [comment]
> Is the khal version supporting this version not yet on 24.11?
> I can't seem to reproduce your issue ```sh $ cat ~/.config/khal/config | grep addresses addresses= <redacted> $ nix shell github:NixOS/nixpkgs/24.11#khal $ khal --version khal, version 0.11.3 $ khal at $ khal printcalendars ... # executes succesfully ``` At first glance your problem stands unrelated to this issue. In any case, it is a better idea to post this question with the resulting `~/.config/khal/config` and the `khal --version`.

the resulting khal config looks like this
```ini
[calendars]
[[default]]
path = /home/malik/.local/share/calendar/default/
addresses= redacted
priority=10
type=calendar



[default]
default_calendar=default
highlight_event_days=true

[locale]
dateformat=%x
datetimeformat=%c
firstweekday=0
longdateformat=%x
longdatetimeformat=%c
timeformat=%X
unicode_symbols=true
weeknumbers=off
```
The email is correct but the spaces seem off

**Pull request merged**